### DISCUSSION TITLE
How does create-theme work?

### DISCUSSION THREAD

**Discussion by ryan-ju** [tamagui/tamagui#835] [Q&A]
I find this [doc](https://tamagui.dev/docs/intro/colors) page very vague on how to create custom theme, specifically, creating just the primary color and derive other colors from it.

How to use @tamagui/create-theme exactly?  It has no exported function to create a theme, and it only takes colors from `import { colorTokens, darkColors, lightColors } from './tokens'`.  Are we supposed to copy the code and add our own color?

**Comment by natew** [comment]
The best docs we have atm is this file:

https://github.com/tamagui/tamagui/blob/master/packages/themes/src/themes.tsx

**Reply by natew** [reply]
I'd recommend copy-pasting that and modifying to your taste

**Reply by ryan-ju** [reply]
Thanks, that's what I figured too.  It would be great to add a function to create consistent themes more easily.

**Reply by natew** [reply]
The lower level package that that uses is a set of functions that makes it pretty simple. We could maybe do a higher level one as well for simple use cases.

**Comment by bailout00** [comment]
I'm struggling with this too, I cannot for the life of me make it work. 

Is there a full-site example somewhere with a custom theme? Customized components?

I'm simply trying to add a specific background color to a component (nav bar) for light/dark.

Some clarifying questions:
When you 'createTheme' does this theme then need to be added to the existing theme or is that done on the tamagui side?
Once the theme is included, let's say I have a styled Component called "NavBarContainer":

```
const dark_NavBarContainer = createTheme({
    background: 'darkred',
    color: 'white',
})

const light_NavBarContainer = createTheme({
    background: 'lightred',
    color: 'black',
})

export const NavBarContainer = styled(Nav, {
    name: "NavBarContainer",
    // I have tried with/without this
   backgroundColor: "$background"
});

// The component

export function NavBar() {
    return (
        <NavBarContainer>
            <Button>Test button</Button>
        </NavBarContainer>
    );
}
```

With this, the background color of NavBarContainer/NavBar is white/non-existant.

Using the example from the docs here, I can't seem to make this work:
https://tamagui.dev/docs/intro/themes

I've tried including the theme in the config, and not including, tried adding custom elements to the theme. I'm using tamagui, not core if that makes a difference. The default light/dark themes are working. 

I'm happy to help update the docs once I understand it, for other's benefit.

**Reply by bailout00** [reply]
This is my config, I took the themes and tokens verbatim from your posted link:

```
import { createTamagui } from "tamagui";
import { config } from "@tamagui/config";
import { tokens } from "./tokens";
import { themes } from "./themes";

const tamaConfig = createTamagui({
    ...config,
    defaultTheme: "light",
    themes,
    tokens,
});

export type Conf = typeof tamaConfig;

declare module "tamagui" {
    interface TamaguiCustomConfig extends Conf {}
}

export default tamaConfig;

```

**Reply by tylerrrkd** [reply]
> I'm struggling with this too, I cannot for the life of me make it work.
> 
> Is there a full-site example somewhere with a custom theme? Customized components?
> 
> I'm simply trying to add a specific background color to a component (nav bar) for light/dark.
> 
> Some clarifying questions: When you 'createTheme' does this theme then need to be added to the existing theme or is that done on the tamagui side? Once the theme is included, let's say I have a styled Component called "NavBarContainer":
> 
> ```
> const dark_NavBarContainer = createTheme({
>     background: 'darkred',
>     color: 'white',
> })
> 
> const light_NavBarContainer = createTheme({
>     background: 'lightred',
>     color: 'black',
> })
> 
> export const NavBarContainer = styled(Nav, {
>     name: "NavBarContainer",
>     // I have tried with/without this
>    backgroundColor: "$background"
> });
> 
> // The component
> 
> export function NavBar() {
>     return (
>         <NavBarContainer>
>             <Button>Test button</Button>
>         </NavBarContainer>
>     );
> }
> ```
> 
> With this, the background color of NavBarContainer/NavBar is white/non-existant.
> 
> Using the example from the docs here, I can't seem to make this work: https://tamagui.dev/docs/intro/themes
> 
> I've tried including the theme in the config, and not including, tried adding custom elements to the theme. I'm using tamagui, not core if that makes a difference. The default light/dark themes are working.
> 
> I'm happy to help update the docs once I understand it, for other's benefit.

@bailout00 hi friend, same question with you!!!!!! #1190
